src/util/builders/url/base.director.ts

```diff
@@ -0,0 +1,44 @@
+import urlBuilder from './url.builder.js'
+
+/**
+ * Interface to build URLs based on base URL
+ */
+export default class BaseURLDirector {
+  private baseDomain: string
+
+  constructor(baseDomain: string) {
+    this.baseDomain = baseDomain
+  }
+
+  buildURL(path: string) {
+    return urlBuilder.setDomain(this.baseDomain).setPath(path).build(true)
+  }
+
+  buildRestApiURL(path?: string) {
+    return urlBuilder
+      .setDomain(this.baseDomain)
+      .setPath(path ?? 'api/rest_v1')
+      .build(true, '/')
+  }
+
+  buildMobileRestApiURL(path?: string) {
+    return urlBuilder
+      .setDomain(this.baseDomain)
+      .setPath(path ?? 'api/rest_v1/page/mobile-sections')
```

Comment:
@kelson42 I'm going to remove all methods related to /mobile-sections endpoint because it has been disabled for a while, you can check the doc - https://en.wikipedia.org/api/rest_v1/#/Mobile/getSections. Pls. let me know, if you have any concerns about it.

Reply:
@VadimKovalenkoSNF Just please articulate/organize your moves (remove mobile-section support, add support of new mobile HTML end-point) with https://github.com/openzim/mwoffliner/issues/1664 (future PR)